shared/logutil/logutil.go

```diff
@@ -28,3 +30,28 @@ func ConfigurePersistentLogging(logFileName string) error {
 	logrus.Info("File logging initialized")
 	return nil
 }
+
+//Masks the url credentials before logging for security purpose
+//[scheme:][//[userinfo@]host][/]path[?query][#fragment] -->  [scheme:][//[***]host][/***][#***]
+//if the format is not matched nothing is done, string is returned as is.
+func MaskCredentialsLogging( currUrl string) string {
+	//error if the input is not a URL
```

Comment:
This function was unit testing with those cases

MaskCredentialsLogging("https://a:b@xyz.net")
MaskCredentialsLogging("https://eth-goerli.alchemyapi.io/v2/tOZG5mjl3.zl_nZdZTNIBUzsDq62R_dkOtY")
MaskCredentialsLogging("https://google.com/search?q=golang")
MaskCredentialsLogging("https://user@example.com/foo%2fbar")
MaskCredentialsLogging("http://john@example.com/#x/y%2Fz")
MaskCredentialsLogging("https://me:pass@example.com/foo/bar?x=1&y=2")